code/game/machinery/computer/camera.dm

```diff
@@ -50,6 +50,8 @@
 
 /obj/machinery/computer/security/ui_interact(mob/user, datum/tgui/ui)
 	. = ..()
+	if(isnull(user.client)) //uncliented mobs should not interact (namely, monkeys)
```

Comment:
There should a better way than this. Have you tried checking the return value of this proc right above this line? It might be false/true when there is no client?

Or a better way is to do this
https://github.com/tgstation/tgstation/blob/ad4bff7687b570a756da4aa97c31850e691e20e3/code/modules/reagents/chemistry/machinery/reagentgrinder.dm#L212-L213

Use the can perform action proc. First test the return value of this proc preferably and then use this

Reply:
The return value above is simply always FALSE, as the procs at levels above this one perform no checks.

As for `can_perform_action`, I could check for dexterity (the ADVANCEDTOOLUSER trait) to stop monkeys in particular from using camera consoles - which seems reasonable to me.

In general, I think it might be good if the game never tries to show UIs to clientless mobs, but I wasn't sure how to go about this on a more fundamental level.

Reply:
> I think it might be good if the game never tries to show UIs to clientless mobs

I agree but this was discovered only when monkeys interact with camera consoles, what about other machines? No where else is this kind of check written so should we write this everywhere else now?

That's what I'm trying to figure out.

Reply:
Ok upon looking further at the camera console code it's one of the few machines that uses the client object not to display the UI but to register the map object

https://github.com/tgstation/tgstation/blob/dbdbf61885964b68f4d66cf2f58d4c1b7e9dfe2e/code/game/machinery/computer/camera.dm#L74

And this is where it runtimes. As long as no other machines uses the `client` var directly it should not matter if it's null or not so this is fine.

> As for can_perform_action, I could check for dexterity (the ADVANCEDTOOLUSER trait) to stop monkeys in particular from using camera consoles - which seems reasonable to me.

This is a good suggestion. If you can do it go ahead